### PR TITLE
fix nil->_xxx crash

### PR DESCRIPTION
**Pull request by wangyuanyuanyy**
fix nil->_xxx crash

### CODE REVIEW

**Comment by joshh-devrel** [comment]
@wangyuanyuanyy Can you resolve the merge conflicts? Thanks!

**Comment by wangyuanyuanyy** [comment]
OK~

File: adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m [hunk 1]
```diff
@@ -56,43 +53,45 @@ - (void)renderBannerForAdConfiguration:(nonnull GADMediationBannerAdConfiguratio
...
-    _loadCompletionHandler(nil, error);
+    self.loadCompletionHandler(nil, error);
     return;
   }
 
   PAGBannerRequest *request = [PAGBannerRequest requestWithBannerSize:bannerSize];
   request.adString = adConfiguration.bidResponse;
-
-  GADPangleRTBBannerRenderer *__weak weakSelf = self;
+  __weak typeof(self) weakSelf = self;
```
Comment (joshh-devrel) [inline adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m:69]: Can you update this to `GADPangleRTBBannerRenderer *__weak weakSelf = self;`? The same applies to other ad formats.

File: adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m [hunk 2]
```diff
@@ -56,43 +53,45 @@ - (void)renderBannerForAdConfiguration:(nonnull GADMediationBannerAdConfiguratio
...
-                strongSelf->_bannerAd = bannerAd;
-                strongSelf->_bannerAd.delegate = strongSelf;
-
-                if (strongSelf->_loadCompletionHandler) {
-                  strongSelf->_delegate = strongSelf->_loadCompletionHandler(strongSelf, nil);
-                }
-              }];
+              completionHandler:^(PAGBannerAd * _Nullable bannerAd,
+                                  NSError * _Nullable loadError) {
+    __strong typeof(weakSelf) strongSelf = weakSelf;
```
Comment (joshh-devrel) [inline adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m:74]: Can you update this to `GADPangleRTBBannerRenderer *strongSelf = weakSelf;`? The same applies to the other ad formats.

File: adapters/Pangle/PangleAdapter/GADMediationAdapterPangleConstants.h [hunk 3]
```diff
@@ -15,6 +15,7 @@
 #import <Foundation/Foundation.h>
 
 /// Pangle adapter version.
+
```
Comment (joshh-devrel) [inline adapters/Pangle/PangleAdapter/GADMediationAdapterPangleConstants.h:18]: Can remove empty space?

File: adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBRewardedRenderer.m [hunk 4]
```diff
@@ -68,51 +68,59 @@ - (void)renderRewardedAdForAdConfiguration:
...
-  id<GADMediationRewardedAdEventDelegate> delegate = _delegate;
-  [delegate didRewardUser];
+  NSNumber *amount =
+      [NSDecimalNumber numberWithInteger:rewardModel.rewardAmount];
+  GADAdReward *reward = [[GADAdReward alloc]
+      initWithRewardType:@""
+            rewardAmount:[NSDecimalNumber decimalNumberWithDecimal:[amount decimalValue]]];
+
+  id<GADMediationRewardedAdEventDelegate> delegate = self.delegate;
+  [delegate didRewardUserWithReward:reward];
```
Comment (joshh-devrel) [inline adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBRewardedRenderer.m:123]: Revert, use the `didRewardUser` API.

File: adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m [hunk 5]
```diff
@@ -21,24 +21,21 @@
 
 @interface GADPangleRTBBannerRenderer () <PAGBannerAdDelegate>
 
+@property (nonatomic, copy) GADMediationBannerLoadCompletionHandler loadCompletionHandler;
+@property (nonatomic, strong) PAGBannerAd *bannerAd;
+@property (nonatomic, weak) id<GADMediationBannerAdEventDelegate> delegate;
+
```
Comment (joshh-devrel) [inline adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m:27]: @wangyuanyuanyy What is the exact error that you saw that led to this change?

Comment (wangyuanyuanyy) [reply adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m:27]: img_v2_e82ca309-01bb-4468-9154-911f6ca6dbeg
 If for some reason “ self ” is released earlier，`self.xxx `results in determining whether the method will be called, but `self->xxx` will be called directly, resulting in a crash.

File: adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m [hunk 6]
```diff
@@ -59,78 +59,78 @@ - (void)renderNativeAdForAdConfiguration:
...
-                 strongSelf->_nativeAd.rootViewController = adConfiguration.topViewController;
-
-                 if (strongSelf->_loadCompletionHandler) {
-                   id<GADMediationNativeAdEventDelegate> delegate =
-                       strongSelf->_loadCompletionHandler(strongSelf, nil);
-                   strongSelf->_delegate = delegate;
-                 }
-               }];
+  
+  PAGLNativeAd *__weak weakSelf = self;
```
Comment (joshh-devrel) [inline adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m:71]: This should be `GADPangleRTBNativeRenderer` instead.

File: adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m [hunk 7]
```diff
@@ -56,43 +53,45 @@ - (void)renderBannerForAdConfiguration:(nonnull GADMediationBannerAdConfiguratio
...
-                if (strongSelf->_loadCompletionHandler) {
-                  strongSelf->_delegate = strongSelf->_loadCompletionHandler(strongSelf, nil);
-                }
-              }];
+              completionHandler:^(PAGBannerAd * _Nullable bannerAd,
+                                  NSError * _Nullable loadError) {
+    GADPangleRTBBannerRenderer *strongSelf = weakSelf;
+    if (!strongSelf) {
+       return;
+    }
```
Comment (joshh-devrel) [inline adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m:77]: This
```
    if (!strongSelf) {
       return;
    }
```
should fix the issue.

Can you revert all the other changes?

Comment (wangyuanyuanyy) [reply adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m:77]: Your means is remove ` if (!strongSelf) {
       return;
    }` ?  Could you tell me the reason?

Comment (joshh-devrel) [reply adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m:77]: I mean we can include this check and revert all the other changes.

Comment (joshh-devrel) [reply adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m:77]: @wangyuanyuanyy What do you think?

Comment (wangyuanyuanyy) [reply adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m]: ok~

**Pull request merged**